### PR TITLE
GH4251/4250/4249/4248/4247: Update dependencies

### PR DESCRIPTION
**Pull request by devlead**
* (GH-4251) Update System.Reflection.Metadata 8.0.0
* (GH-4250) Update System.Collections.Immutable to 8.0.0
* (GH-4249) Update Microsoft.Extensions.DependencyInjection to 8.0.0
* (GH-4248) Update Microsoft.NET.Test.Sdk to 17.8.0
* (GH-4247) Update xunit to 2.6.1

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,6 +25,11 @@ jobs:
         with:
           dotnet-version: '6.0.x'
 
+      - name: Install .NET SDK 7.0.x
```
Comment (pascalberger) [inline .github/workflows/build.yml:28]: Why is this required?

**Review by pascalberger — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (devlead) [reply .github/workflows/build.yml:28]: It installs latest .NET 7 SDK as global json will install .NET 8.0.x SDK,

But it broke many things, so will revert and take it with .NET 8 support PR.

**Review by pascalberger — APPROVED** [review]
